test/e2e/storage/external/external.go

```diff
@@ -302,6 +312,42 @@ func (d *driverDefinition) GetDynamicProvisionStorageClass(e2econfig *testsuites
 	return testsuites.GetStorageClass(sc.Provisioner, sc.Parameters, sc.VolumeBindingMode, f.Namespace.Name, "e2e-sc")
 }
 
+func (d *driverDefinition) GetTimeouts() *framework.TimeoutContext {
+	timeouts := framework.NewTimeoutContextWithDefaults()
+	if d.Timeouts == nil {
+		return timeouts
+	}
+
+	// Use a temporary map to hold the timeouts specified in the manifest file
+	c := make(map[string]time.Duration)
+	for k, v := range d.Timeouts {
+		duration, err := time.ParseDuration(v)
+		if err != nil {
+			// We can't use ExpectNoError() because his method can be called out of an It(),
+			// so we simply log the error and return the default timeouts.
+			klog.Errorf("Could not parse duration for key %s: %v", k, err)
+			return timeouts
+		}
+		c[k] = duration
+	}
+
+	// Convert the temporary map holding the custom timeouts to JSON
+	t, err := json.Marshal(c)
+	if err != nil {
+		klog.Errorf("Could not marshal timeouts: %v", err)
+		return timeouts
+	}
+
+	// Override the default timeouts with the custom ones
+	err = json.Unmarshal(t, &timeouts)
+	if err != nil {
```

Comment:
just want to make sure at this point, the returned timeouts will still be the default value?
also in the log, maybe mention returning default timeout?

Reply:
Right, `timeouts` contains the default values at this point.

Extended log message...